### PR TITLE
misc(proto): ensure all strings are well-formed

### LINKED ISSUE #15908: generic::internal: Error unmarshalling JSON into proto:  "lighthouseVersion":"11.5.0"
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/main/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/main/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://tangrammontessori.site/

### What happened?

Getting error while checking Page Speed. 

### What did you expect?

A Report 

### What have you tried?

https://pagespeed.web.dev/analysis/http-tangrammontessori-site/kvkrh86eqs?form_factor=desktop

### How were you running Lighthouse?

PageSpeed Insights

### Lighthouse Version

11.50.0

### Chrome Version

_No response_

### Node Version

_No response_

### OS

_No response_

### Relevant log output

```shell
Oops! Something went wrong.
generic::internal: Error unmarshalling JSON into proto: {"lighthouseVersion":"11.5.0","requestedUrl":"https://tangrammontessori.site/","mainDocumentUrl":"https://tangrammontessori.site/","finalDisplayedUrl":"https://tangrammontessori.site/","finalUrl":"https://tangrammontessori.site/","fetchTime":"2024-04-02T13:49:16.763Z","gatherMode":"navigation","runWarnings":[],"userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/122.0.6261.94 Safari/537.36","environment":{"networkUserAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36","hostUserAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/122.0.6261.94 Safari/537.36","benchmarkIndex":359,"credits":{"axe-core":"4.8.1"}},"audits":{"is-on-https":{"id":"is-on-https","title":"Uses HTTPS","description":"All sites should be protected with HTTPS, even ones that don't handle sensitive data. This includes avoiding [mixed content](https://developers.google.com/web/fundamentals/security/prevent-mixed-content/what-is-mixed-content), where some resources are loaded over HTTP despite the initial request being served over HTTPS. HTTPS prevents intruders from tampering with or passively listening in on the communications between your app and your users, and is a prerequisite for HTTP/2 and many new web platform APIs. [Learn more about HTTPS](https://developer.chrome.com/docs/lighthouse/pwa/is-on-https/)."
```

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #15908

ref #14911

### CODE REVIEW

File: core/lib/proto-preprocessor.js [hunk 1]
```diff
@@ -83,28 +83,44 @@ function processForProto(lhr) {
...
         if (typeof item === 'object' || Array.isArray(item)) {
-          removeStrings(item);
+          iterateStrings(item, cb);
         }
       });
     }
   }
 
-  removeStrings(reportJson);
+  iterateStrings(reportJson, (obj, key) => {
```
Comment (paulirish) [inline core/lib/proto-preprocessor.js:108]: maybe extract this cb into a fn with a name?   `removeEmptyAndDropLoneSurrogates` ?

File: core/lib/proto-preprocessor.js [hunk 2]
```diff
@@ -83,28 +83,44 @@ function processForProto(lhr) {
...
+
+    // Remove empty strings, as they are dropped after round-tripping anyway.
+    if (value === '') {
+      delete obj[key];
+      return;
+    }
+
+    // Sanitize lone surrogates.
+    // @ts-expect-error node 20
+    if (String.prototype.isWellFormed && !value.isWellFormed()) {
```
Comment (adamraine) [inline core/lib/proto-preprocessor.js:119]: Could we throw an error if `String.prototype.isWellFormed` isn't available? Seems like we would want to fail loudly in that case.

Comment (adamraine) [reply core/lib/proto-preprocessor.js:119]: Gotcha, no error sounds good

**Review by adamraine — APPROVED** [review]

File: core/lib/proto-preprocessor.js [hunk 3]
```diff
@@ -83,28 +83,44 @@ function processForProto(lhr) {
...
+          cb(obj, key);
+        } else {
+          iterateStrings(obj[key], cb);
         }
       });
     } else if (Array.isArray(obj)) {
       obj.forEach(item => {
         if (typeof item === 'object' || Array.isArray(item)) {
-          removeStrings(item);
+          iterateStrings(item, cb);
```
Comment (brendankenny) [inline core/lib/proto-preprocessor.js:102]: Technically this allows arrays with non-well-formed strings in them to pass through unchanged.

**Pull request merged**